lib/wda-mac.js

```diff
@@ -278,19 +284,55 @@ class WDAMacServer {
     }
   }
 
+
+  /**
+   * @typedef {Object} ProxyToInfo
+   *
+   * @property {string} scheme - The scheme proxy to.
+   * @property {string} host - The host name proxy to.
+   * @property {number} port - The port number proxy to.
+   * @property {string} path - The path proxy to.
+   */
+
+  /**
+   * Returns information about proxy information which WDAMacServer proxy to.
+   *
+   * @param {Object} caps - The capabilities provided by a client
+   * @return {ProxyToInfo}
+   */
+  getProxyToInfo (caps) {
+    let scheme = 'http';
+    if (_.isString(caps.webDriverAgentMacUrl)) {
+      const {protocol, hostname, port, pathname} = url.parse(caps.webDriverAgentMacUrl);
+      if (_.isString(protocol)) {
+        scheme = protocol.split(':')[0];
+      }
+      return {
+        scheme,
+        host: hostname ?? HOST,
+        port: _.isNull(port) ? DEFAULT_SYSTEM_PORT : _.parseInt(port),
+        path: pathname === '/' ? '' : pathname
+      };
+    }
+
+    return {scheme, host: HOST, port: this.process.port, path: ''};
+  }
+
   async startSession (caps) {
     this.serverStartupTimeoutMs = caps.serverStartupTimeout ?? this.serverStartupTimeoutMs;
     const wasProcessInitNecessary = await this.process.init(caps);
 
     if (wasProcessInitNecessary || !(await this.isProxyReady(false))) {
+      const {scheme, server, port, path} = this.getProxyToInfo(caps);
       this.proxy = new WDAMacProxy({
-        server: HOST,
-        port: this.process.port,
-        base: '',
+        scheme,
+        server,
+        port,
+        base: path,
         keepAlive: true,
       });
       this.proxy.didProcessExit = false;
-      this.process.proc.on('exit', () => {
+      this.process.proc?.on('exit', () => {
```

Comment:
process.proc must always be defined after calling `init`. It is fine to throw an exception here if proc is undefined